gpjax/linops/linear_operator.py

```diff
@@ -60,58 +56,58 @@ def ndim(self) -> int:
         return len(self.shape)
 
     @property
-    def T(self) -> LinearOperator:
+    def T(self) -> "LinearOperator":
         """Transpose linear operator. Currently, we assume all linear operators are square and symmetric."""
         return self
 
     def __sub__(
-        self, other: Union[LinearOperator, Float[Array, "N N"]]
-    ) -> LinearOperator:
+        self, other: Union["LinearOperator", Float[Array, "N N"]]
+    ) -> "LinearOperator":
         """Subtract linear operator."""
         return self + (other * -1)
 
     def __rsub__(
-        self, other: Union[LinearOperator, Float[Array, "N N"]]
-    ) -> LinearOperator:
+        self, other: Union["LinearOperator", Float[Array, "N N"]]
+    ) -> "LinearOperator":
         """Reimplimentation of subtract linear operator."""
         return (self * -1) + other
 
     def __add__(
-        self, other: Union[LinearOperator, Float[Array, "N N"]]
-    ) -> LinearOperator:
+        self, other: Union["LinearOperator", Float[Array, "N N"]]
+    ) -> "LinearOperator":
         """Add linear operator."""
         raise NotImplementedError
 
     def __radd__(
-        self, other: Union[LinearOperator, Float[Array, "N N"]]
-    ) -> LinearOperator:
+        self, other: Union["LinearOperator", Float[Array, "N N"]]
+    ) -> "LinearOperator":
         """Reimplimentation of add linear operator."""
         return self + other
 
     @abc.abstractmethod
-    def __mul__(self, other: float) -> LinearOperator:
+    def __mul__(self, other: ScalarFloat) -> "LinearOperator":
         """Multiply linear operator by scalar."""
         raise NotImplementedError
 
-    def __rmul__(self, other: float) -> LinearOperator:
+    def __rmul__(self, other: ScalarFloat) -> "LinearOperator":
         """Reimplimentation of multiply linear operator by scalar."""
         return self.__mul__(other)
 
     @abc.abstractmethod
-    def _add_diagonal(self, other: DiagonalLinearOperator) -> LinearOperator:
+    def _add_diagonal(self, other: "gpjax.linops.diagonal_linear_operator.DiagonalLinearOperator") -> "LinearOperator":
```

Comment:
Do we need `gpjax.linops.diagonal_linear_operator`?

Reply:
yes, because it's a reference to a class in a different module